WordPress/src/main/java/org/wordpress/android/viewmodel/PluginBrowserViewModel.java

```diff
@@ -0,0 +1,395 @@
+package org.wordpress.android.viewmodel;
+
+import android.app.Application;
+import android.arch.lifecycle.AndroidViewModel;
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.os.Handler;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.fluxc.Dispatcher;
+import org.wordpress.android.fluxc.generated.PluginActionBuilder;
+import org.wordpress.android.fluxc.model.SiteModel;
+import org.wordpress.android.fluxc.model.plugin.PluginDirectoryType;
+import org.wordpress.android.fluxc.model.plugin.SitePluginModel;
+import org.wordpress.android.fluxc.model.plugin.WPOrgPluginModel;
+import org.wordpress.android.fluxc.store.PluginStore;
+import org.wordpress.android.ui.plugins.PluginBrowserActivity;
+import org.wordpress.android.ui.plugins.PluginUtils;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class PluginBrowserViewModel extends AndroidViewModel {
+    public enum PluginListStatus {
+        CAN_LOAD_MORE,
+        DONE,
+        ERROR,
+        FETCHING,
+        LOADING_MORE
+    }
+
+    @Inject Dispatcher mDispatcher;
+    @Inject PluginStore mPluginStore;
+
+    private String mSearchQuery;
+    private SiteModel mSite;
+
+    private final Handler mHandler;
+
+    private MutableLiveData<PluginListStatus> mNewPluginsListStatus;
```

Comment:
Fixed in 913bd2531ccb4dd8a7f8adbc80abf77e98c4387d. I've also added `@SuppressWarnings("WeakerAccess")` to the constructor because we need it to be `public` for the view model to work.